src/crawlers/browser_crawler.js

```diff
@@ -390,6 +390,10 @@ export default class BrowserCrawler extends BasicCrawler {
         tryCancel();
         this._enhanceCrawlingContextWithPageInfo(crawlingContext, page, useIncognitoPages);
 
+        // DO NOT MOVE THIS LINE ABOVE!
+        // `enhanceCrawlingContextWithPageInfo` can (and will!) override those properties.
```

Comment:
Should this comment be more detailed? Example: the session in this scenario can be used only once, so if we save the session before making sure it was used only once, we get that error.

Reply:
Yes, please add some practical context. There might come a moment when someone needs to refactor this function and it would really help them to know what should not be happening after they make the changes.

Reply:
Yeah a bit more context would be good, but nothing too verbose. Maybe just describe what side effects it can have.